src/plugins/vis_builder/public/visualizations/table/components/table_viz_options.tsx

```diff
@@ -0,0 +1,227 @@
+    text: i18n.translate('visTypeTableNew.totalAggregations.sumText', {
+      defaultMessage: 'Sum',
+    }),
+  },
+  {
+    value: AggTypes.AVG,
+    text: i18n.translate('visTypeTableNew.totalAggregations.averageText', {
+      defaultMessage: 'Average',
+    }),
+  },
+  {
+    value: AggTypes.MIN,
+    text: i18n.translate('visTypeTableNewNew.totalAggregations.minText', {
+      defaultMessage: 'Min',
+    }),
+  },
+  {
+    value: AggTypes.MAX,
+    text: i18n.translate('visTypeTableNewNew.totalAggregations.maxText', {
+      defaultMessage: 'Max',
+    }),
+  },
+  {
+    value: AggTypes.COUNT,
+    text: i18n.translate('visTypeTableNewNew.totalAggregations.countText', {
+      defaultMessage: 'Count',
+    }),
+  },
+];
+
+function TableVizOptions() {
+  const styleState = useTypedSelector((state) => state.style) as TableOptionsDefaults;
+  const { aggConfigs } = useAggs();
+  const dispatch = useTypedDispatch();
+
+  const setOption = useCallback(
+    (callback: (draft: Draft<typeof styleState>) => void) => {
+      const newState = produce(styleState, callback);
+      dispatch(setStyleState<TableOptionsDefaults>(newState));
+    },
+    [dispatch, styleState]
+  );
+
+  const percentageColumns = useMemo(() => {
+    const defaultPercentageColText = {
+      value: '',
+      text: i18n.translate('visTypeTableNew.params.defaultPercentageCol', {
+        defaultMessage: 'Don’t show',
+      }),
+    };
+    return aggConfigs
+      ? [
+          defaultPercentageColText,
+          ...tabifyGetColumns(aggConfigs.getResponseAggs(), true)
+            .filter((col) => get(col.aggConfig.toSerializedFieldFormat(), 'id') === 'number')
+            .map(({ name }) => ({ value: name, text: name })),
+        ]
+      : [defaultPercentageColText];
+  }, [aggConfigs]);
+
+  useEffect(() => {
+    if (
+      !percentageColumns.find(({ value }) => value === styleState.percentageCol) &&
+      percentageColumns[0] &&
+      percentageColumns[0].value !== styleState.percentageCol
+    ) {
+      setOption((draft) => {
+        draft.percentageCol = percentageColumns[0].value;
+      });
+    }
+  }, [percentageColumns, styleState.percentageCol, setOption]);
+
+  const isPerPageValid = styleState.perPage === '' || styleState.perPage > 0;
+
+  return (
+    <>
+      <Option
+        title={i18n.translate('visTypeTableNewNew.params.settingsTitle', {
+          defaultMessage: 'Settings',
+        })}
+        initialIsOpen
+      >
+        <NumberInputOption
+          label={
+            <>
+              {i18n.translate('visTypeTableNewNew.params.perPageLabel', {
+                defaultMessage: 'Max rows per page',
+              })}
+              ,
+              <EuiIconTip
+                content={
+                  <FormattedMessage
+                    id="visTypeTableNewNews.field.emptyTooltip"
+                    defaultMessage="Leaving this field empty means it will use number of buckets from the response."
```

Comment:
should it just be "number of rows"? Also, the pagination seems to behave weirdly if it is removed/left blank. My expectation would be that leaving it blank would simply disable pagination altogether.

Reply:
I am using the original contents from old table [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d7004dc5b0392477fdd54ac66b29d231975a173b/src/plugins/vis_type_table/public/components/table_vis_options.tsx#L92)

```
Leaving this field empty means it will use number of buckets from the response.
```